crates/router/src/core/files.rs

```
@@ -0,0 +1,120 @@
+        file_upload_provider: Some(file_upload_provider.foreign_into()),
+        available: true,
+    };
+    state
+        .store
+        .update_file_metadata(file_metadata_object, update_file_metadata)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable_lazy(|| {
+            format!("Unable to update file_metadata with file_id: {}", file_id)
+        })?;
+    Ok(services::api::ApplicationResponse::Json(
+        files::CreateFileResponse { file_id },
+    ))
+}
+
+pub async fn files_delete_core(
+    state: &AppState,
+    merchant_account: storage::MerchantAccount,
+    req: api::FileId,
+) -> RouterResponse<serde_json::Value> {
+    helpers::delete_file_using_file_id(state, req.file_id.clone(), &merchant_account).await?;
+    state
+        .store
+        .delete_file_metadata_by_merchant_id_file_id(&merchant_account.merchant_id, &req.file_id)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Unable to delete file_metadata")?;
+    Ok(ApplicationResponse::StatusOk)
+}
+
+pub async fn files_retrieve_core(
+    state: &AppState,
+    merchant_account: storage::MerchantAccount,
+    req: api::FileId,
+) -> RouterResponse<serde_json::Value> {
+    let file_metadata_object = state
+        .store
+        .find_file_metadata_by_merchant_id_file_id(&merchant_account.merchant_id, &req.file_id)
+        .await
+        .change_context(errors::ApiErrorResponse::FileNotFound)
+        .attach_printable("Unable to retrieve file_metadata")?;
+    let (received_data, _provider_file_id) =
+        helpers::retrieve_file_and_provider_file_id_from_file_id(
+            state,
+            Some(req.file_id),
+            &merchant_account,
+        )
+        .await?;
+    let content_type = file_metadata_object
+        .file_type
+        .parse::<mime::Mime>()
+        .map_err(|_| errors::ApiErrorResponse::InternalServerError)
```

Don't discard the error